diff --git a/plugins/FeedsFileProcessor.inc b/plugins/FeedsFileProcessor.inc
index 42c25f9..7234922 100644
--- a/plugins/FeedsFileProcessor.inc
+++ b/plugins/FeedsFileProcessor.inc
@@ -234,6 +234,9 @@ class FeedsFileProcessor extends FeedsProcessor {
     // There is no body field on files.
     unset($form['input_format']);
 
+    // Add 'undefined' option to allow auto-detection of file types.
+    $form['bundle']['#options'] = array(FILE_TYPE_NONE => t('Undefined')) + $form['bundle']['#options'];
+
     $author = user_load($this->config['author']);
     $form['author'] = array(
       '#type' => 'textfield',
