diff --git a/plugins/FeedsFileProcessor.inc b/plugins/FeedsFileProcessor.inc
index 42c25f9..a06f2e1 100644
--- a/plugins/FeedsFileProcessor.inc
+++ b/plugins/FeedsFileProcessor.inc
@@ -84,7 +84,9 @@ class FeedsFileProcessor extends FeedsProcessor {
    *   If validation fails.
    */
   protected function entityValidate($entity) {
-
+    if (empty($entity->uri)) {
+      throw new FeedsValidationException(t('Missing file URI.'));
+    }
   }
 
   /**
