diff --git a/panels_ipe/plugins/display_renderers/panels_renderer_ipe.class.php b/panels_ipe/plugins/display_renderers/panels_renderer_ipe.class.php
index 6e011f7..767a925 100644
--- a/panels_ipe/plugins/display_renderers/panels_renderer_ipe.class.php
+++ b/panels_ipe/plugins/display_renderers/panels_renderer_ipe.class.php
@@ -264,6 +264,11 @@ class panels_renderer_ipe extends panels_renderer_editor {
     if (!empty($form_state['clicked_button']['#save-display'])) {
       // Saved. Save the cache.
       panels_edit_cache_save($this->cache);
+      // Clear the cache and make sure the current display reflects the most
+      // recent data that was saved.
+      panels_edit_cache_clear($this->cache);
+      $this->cache = panels_edit_cache_get($this->display->cache_key);
+      $this->display = $this->cache->display;
       // A rerender should fix IDs on added panes as well as ensure style changes are
       // rendered.
       $this->meta_location = 'inline';
