diff --git a/commerce_coupon.install b/commerce_coupon.install
index 1f313b6..647bbac 100644
--- a/commerce_coupon.install
+++ b/commerce_coupon.install
@@ -131,9 +131,12 @@ function commerce_coupon_uninstall() {
   // Delete all fields that belong to the module's entity types.
   commerce_delete_instances('commerce_coupon_log');
   commerce_delete_instances('commerce_coupon');
+  commerce_delete_field('commerce_coupon_number_of_uses');
+  commerce_delete_field('commerce_coupon_code');
   commerce_delete_field('commerce_coupon_order_reference');
   commerce_delete_field('commerce_coupon_reference');
-
+  field_purge_batch(20);
+  
   // Delete all module variables.
   variable_del('commerce_coupon_default_code_size');
   variable_del('commerce_coupon_checkout_pane_view');
diff --git a/commerce_coupon.module b/commerce_coupon.module
index bda80a2..f85a046 100644
--- a/commerce_coupon.module
+++ b/commerce_coupon.module
@@ -373,7 +373,10 @@ function commerce_coupon_type_delete_multiple(array $coupon_type_names) {
  */
 function commerce_coupon_type_disable($coupon_type_name) {
   $coupon_type = commerce_coupon_type_load($coupon_type_name);
-  if (!empty($coupon_type)) {
+  $coupons = commerce_coupon_load_multiple(array(), array('type' => $coupon_type->type));
+  
+  // We need to be sure there are existing coupons which to save as inactive.
+  if (!empty($coupon_type) && !empty($coupons)) {
     // Disable the coupon type.
     $coupon_type->status = 0;
     commerce_coupon_type_save($coupon_type, TRUE);
