diff --git a/modules/tax/commerce_tax.module b/modules/tax/commerce_tax.module
index bb56e77e..5d01563d 100644
--- a/modules/tax/commerce_tax.module
+++ b/modules/tax/commerce_tax.module
@@ -199,7 +199,8 @@ function commerce_tax_rates_reset() {
  */
 function commerce_tax_rate_load($name) {
   $tax_rates = commerce_tax_rates();
-  return empty($tax_rates[$name]) ? FALSE : $tax_rates[$name];
+  $tax_name = is_array($name) ? $name[0] : $name;
+  return empty($tax_rates[$tax_name]) ? FALSE : $tax_rates[$tax_name];
 }
 
 /**
