diff --git a/follow.inc b/follow.inc
index 64bf244..718c75c 100644
--- a/follow.inc
+++ b/follow.inc
@@ -5,6 +5,11 @@
  * Follow module API and helper functions.
  */
 
+/* DOMAIN ACCESS */
+$current_domain_info = domain_get_domain();
+$domain_slg = 'domin_' . $current_domain_info['domain_id'];
+define('FOLLOW_DOMAIN', $domain_slg);
+
 define('FOLLOW_NAME', 0);
 define('FOLLOW_ME', 1);
 define('FOLLOW_US', 2);
@@ -80,7 +85,7 @@ function follow_save_css($reset = FALSE) {
 function follow_link_title($uid = 0) {
   // Check to see if we have a valid username.
   if ($uid) {
-    $setting = variable_get('follow_user_block_title', FOLLOW_NAME);
+    $setting = variable_get(FOLLOW_DOMAIN . '_follow_user_block_title', FOLLOW_NAME);
     // Special handling for usernames.
     if ($setting == FOLLOW_NAME) {
       $account = user_load($uid);
@@ -90,9 +95,9 @@ function follow_link_title($uid = 0) {
     return t('Follow me on');
   }
 
-  switch(variable_get('follow_site_block_title', FOLLOW_NAME)) {
+  switch(variable_get(FOLLOW_DOMAIN . '_follow_site_block_title', FOLLOW_NAME)) {
     case FOLLOW_NAME:
-      return t('Follow @name on', array('@name' => variable_get('site_name', 'Drupal')));
+      return t('Follow @name on', array('@name' => variable_get(FOLLOW_DOMAIN . '_site_name', 'Drupal')));
 
     case FOLLOW_ME:
       return t('Follow me on');
@@ -127,10 +132,13 @@ function _follow_links_element(array $links, array $networks, $type, $alignment
   $wrapper = $alignment == 'horizontal' ? 'span' : 'div';
 
   foreach ($links as $link) {
-    $title = !empty($link->title) ? $link->title : $networks[$link->name]['title'];
+    $title = !empty($link->title) ? $link->title : isset($networks[$link->name]['title']) ? $networks[$link->name]['title'] : '';
+
+    //Remove domain from the name to use regular css rules
+    $link_name = str_replace(FOLLOW_DOMAIN . '_', '', $link->name);
 
     $element['follow-links'][$link->name] = array(
-      '#prefix' => "<{$wrapper} class='follow-link-wrapper follow-link-wrapper-{$link->name}'>",
+      '#prefix' => "<{$wrapper} class='follow-link-wrapper follow-link-wrapper-{$link_name}'>",
       '#theme' => 'follow_link',
       '#link' => $link,
       '#title' => $title,
@@ -233,6 +241,7 @@ function follow_build_url_regex($network_info) {
 
   if (!empty($network_info['domain'])) {
     // An external link.
+    $network_info['domain'] = str_replace(FOLLOW_DOMAIN . '_', '', $network_info['domain']);
     return '@^https?://([a-z0-9\-_.]+\\.|)' . str_replace('.', '\\.', $network_info['domain']) . '/@i';
   }
   // An internal link should not have ':'.
@@ -250,8 +259,8 @@ function follow_build_url_regex($network_info) {
 function follow_links_load($uid = 0) {
   $links = array();
 
-  $sql = "SELECT * FROM {follow_links} WHERE uid = :uid ORDER BY weight ASC";
-  $result = db_query($sql, array(':uid' => $uid));
+  $sql = "SELECT * FROM {follow_links} WHERE uid = :uid AND name LIKE :name ORDER BY weight ASC";
+  $result = db_query($sql, array(':uid' => $uid, ':name' => FOLLOW_DOMAIN . '%'));
 
   foreach ($result as $link) {
     $link->options = unserialize($link->options);
@@ -291,6 +300,7 @@ function follow_link_save($link) {
 function follow_link_delete($lid) {
   db_delete('follow_links')
     ->condition('lid', $lid)
+    ->condition('name', FOLLOW_DOMAIN .'%', 'LIKE')
     ->execute();
 }
 
@@ -333,81 +343,81 @@ function follow_networks_load($uid = 0, $reset = FALSE) {
  */
 function follow_default_networks($uid) {
   $networks = array(
-    'facebook'  => array(
+    FOLLOW_DOMAIN . '_facebook'  => array(
       'title' => t('Facebook'),
       'domain' => 'facebook.com',
     ),
-    'googleplus'    => array(
+    FOLLOW_DOMAIN . '_googleplus'    => array(
       'title' => t('Google+'),
       'domain' => 'plus.google.com',
     ),
-    'virb'      => array(
+    FOLLOW_DOMAIN . '_virb'      => array(
       'title' => t('Virb'),
       'domain' => 'virb.com',
     ),
-    'myspace'   => array(
+    FOLLOW_DOMAIN . '_myspace'   => array(
       'title' => t('MySpace'),
       'domain' => 'myspace.com',
     ),
-    'twitter'   => array(
+    FOLLOW_DOMAIN . '_twitter'   => array(
       'title' => t('Twitter'),
       'domain' => 'twitter.com',
     ),
-    'picasa'    => array(
+    FOLLOW_DOMAIN . '_picasa'    => array(
       'title' => t('Picasa'),
       'domain' => 'picasaweb.google.com',
     ),
-    'flickr'    => array(
+    FOLLOW_DOMAIN . '_flickr'    => array(
       'title' => t('Flickr'),
       'domain' => 'flickr.com',
     ),
-    'youtube'   => array(
+    FOLLOW_DOMAIN . '_youtube'   => array(
       'title' => t('YouTube'),
       'domain' => 'youtube.com',
     ),
-    'vimeo'     => array(
+    FOLLOW_DOMAIN . '_vimeo'     => array(
       'title' => t('Vimeo'),
       'domain' => 'vimeo.com',
     ),
-    'bliptv'    => array(
+    FOLLOW_DOMAIN . '_bliptv'    => array(
       'title' => t('blip.tv'),
       'domain' => 'blip.tv',
     ),
-    'lastfm'    => array(
+    FOLLOW_DOMAIN . '_lastfm'    => array(
       'title' => t('last.fm'),
       'domain' => 'last.fm',
     ),
-    'linkedin'  => array(
+    FOLLOW_DOMAIN . '_linkedin'  => array(
       'title' => t('LinkedIn'),
       'domain' => 'linkedin.com',
     ),
-    'delicious' => array(
+    FOLLOW_DOMAIN . '_delicious' => array(
       'title' => t('Delicious'),
       'domain' => 'delicious.com',
     ),
-    'tumblr'    => array(
+    FOLLOW_DOMAIN . '_tumblr'    => array(
       'title' => t('Tumblr'),
       'domain' => 'tumblr.com',
     ),
-    'viadeo'    => array(
+    FOLLOW_DOMAIN . '_viadeo'    => array(
       'title' => t('Viadeo'),
       'domain' => 'viadeo.com',
     ),
-    'xing'    => array(
+    FOLLOW_DOMAIN . '_xing'    => array(
       'title' => t('Xing'),
       'domain' => 'xing.com',
     ),
-    'spiceworks'    => array(
+    FOLLOW_DOMAIN . '_spiceworks'    => array(
       'title' => t('Spiceworks'),
       'domain' => 'spiceworks.com',
     ),
   );
   if ($uid == 0) {
-    $networks['this-site'] = array(
+    $networks[FOLLOW_DOMAIN . '_this-site'] = array(
       'title' => t('This site (RSS)'),
       'domain' => '',
     );
-    $networks['newsletter'] = array(
+    $networks[FOLLOW_DOMAIN . '_newsletter'] = array(
       'title' => t('Newsletter'),
       'domain' => '',
     );
diff --git a/follow.module b/follow.module
index b43aad3..6d805bb 100644
--- a/follow.module
+++ b/follow.module
@@ -203,29 +203,29 @@ function follow_block_info() {
 function follow_block_configure($delta = '') {
   switch($delta) {
     case 'site':
-      $form['follow_title'] = array(
+      $form[FOLLOW_DOMAIN . '_follow_title'] = array(
         '#type' => 'radios',
         '#title' => t('Default block title'),
-        '#default_value' => variable_get('follow_site_block_title', FOLLOW_NAME),
+        '#default_value' => variable_get(FOLLOW_DOMAIN . '_follow_site_block_title', FOLLOW_NAME),
         '#options' => array(
-          FOLLOW_NAME => t('Follow @name on', array('@name' => variable_get('site_name', 'Drupal'))),
+          FOLLOW_NAME => t('Follow @name on', array('@name' => variable_get(FOLLOW_DOMAIN . '_site_name', 'Drupal'))),
           FOLLOW_ME => t('Follow me on'),
           FOLLOW_US => t('Follow us on'),
         ),
       );
-      $form['follow_user'] = array(
+      $form[FOLLOW_DOMAIN . '_follow_user'] = array(
         '#type' => 'checkbox',
         '#title' => t('User pages'),
         '#description' => t('Should this block display on user profile pages?'),
-        '#default_value' => variable_get('follow_site_block_user', TRUE),
+        '#default_value' => variable_get(FOLLOW_DOMAIN . '_follow_site_block_user', TRUE),
       );
       break;
 
     case 'user':
-      $form['follow_title'] = array(
+      $form[FOLLOW_DOMAIN . '_follow_title'] = array(
         '#type' => 'radios',
         '#title' => t('Default block title'),
-        '#default_value' => variable_get('follow_user_block_title', FOLLOW_NAME),
+        '#default_value' => variable_get(FOLLOW_DOMAIN . '_follow_user_block_title', FOLLOW_NAME),
         '#options' => array(
           FOLLOW_NAME => t('Follow [username] on'),
           FOLLOW_ME => t('Follow me on'),
@@ -269,9 +269,9 @@ function follow_block_configure($delta = '') {
  * Implements hook_block_save().
  */
 function follow_block_save($delta = '', $edit = array()) {
-  variable_set("follow_{$delta}_block_title", $edit['follow_title']);
+  variable_set(FOLLOW_DOMAIN . "_follow_{$delta}_block_title", $edit[FOLLOW_DOMAIN . '_follow_title']);
   if ($delta == 'site') {
-    variable_set('follow_site_block_user', $edit['follow_user']);
+    variable_set(FOLLOW_DOMAIN . '_follow_site_block_user', $edit[FOLLOW_DOMAIN . '_follow_user']);
   }
   variable_set("follow_{$delta}_alignment", $edit['follow_alignment']);
   variable_set("follow_{$delta}_hide_text", $edit['follow_hide_text']);
@@ -288,7 +288,7 @@ function follow_block_view($delta = '') {
   switch ($delta) {
     case 'site':
       if (($content = _follow_block_content($type))
-          && (variable_get('follow_site_block_user', TRUE) || !(arg(0) == 'user' && is_numeric(arg(1))))) {
+          && (variable_get(FOLLOW_DOMAIN . '_follow_site_block_user', TRUE) || !(arg(0) == 'user' && is_numeric(arg(1))))) {
         return array(
           'subject' => _follow_block_subject(),
           'content' => $content,
@@ -394,6 +394,7 @@ function theme_follow_link($variables) {
   $title = $variables['title'];
   $classes = array();
   $classes[] = 'follow-link';
+  $link->name = str_replace(FOLLOW_DOMAIN . '_', '', $link->name);
   $classes[] = "follow-link-{$link->name}";
   $classes[] = $link->uid ? 'follow-link-user' : 'follow-link-site';
   $attributes = array(
@@ -457,7 +458,7 @@ function follow_links_form($form, &$form_state, $uid = 0) {
   $form = array();
 
   $form['uid'] = array('#type' => 'hidden', '#value' => $uid);
-  $form['follow_links']['#tree'] = TRUE;
+  $form[FOLLOW_DOMAIN . '_follow_links']['#tree'] = TRUE;
 
   // List all the available links.
   $links = follow_links_load($uid);
@@ -467,7 +468,7 @@ function follow_links_form($form, &$form_state, $uid = 0) {
   if (is_array($links)) {
     foreach ($links as $name => $link) {
       $title = $networks[$name]['title'];
-      $form['follow_links'][$name] = _follow_links_form_link($link, $title, $uid);
+      $form[FOLLOW_DOMAIN . '_follow_links'][$name] = _follow_links_form_link($link, $title, $uid);
       // Unset this specific network so we don't add the same one again below.
       unset($networks[$name]);
     }
@@ -476,7 +477,7 @@ function follow_links_form($form, &$form_state, $uid = 0) {
   foreach ($networks as $name => $info) {
     $link = new stdClass();
     $link->name = $name;
-    $form['follow_links'][$name] = _follow_links_form_link($link, $info['title'], $uid);
+    $form[FOLLOW_DOMAIN . '_follow_links'][$name] = _follow_links_form_link($link, $info['title'], $uid);
   }
 
   $form['submit'] = array('#type' => 'submit', '#value' => t('Submit'));
@@ -528,7 +529,7 @@ function _follow_links_form_link($link, $title, $uid) {
  */
 function follow_links_form_submit($form, &$form_state) {
   $values = $form_state['values'];
-  $links = $values['follow_links'];
+  $links = $values[FOLLOW_DOMAIN . '_follow_links'];
 
   foreach($links as $name => $link) {
     $link = (object) $link;
@@ -565,19 +566,19 @@ function theme_follow_links_form($variables) {
   $rows = array();
   $disabled_rows = array();
 
-  foreach (element_children($form['follow_links']) as $key) {
+  foreach (element_children($form[FOLLOW_DOMAIN . '_follow_links']) as $key) {
     $row = array();
 
-    if (isset($form['follow_links'][$key]['weight'])) {
-      $row[] = drupal_render($form['follow_links'][$key]['lid']) . drupal_render($form['follow_links'][$key]['name']);
-      $row[] = drupal_render($form['follow_links'][$key]['url']);
+    if (isset($form[FOLLOW_DOMAIN . '_follow_links'][$key]['weight'])) {
+      $row[] = drupal_render($form[FOLLOW_DOMAIN . '_follow_links'][$key]['lid']) . drupal_render($form[FOLLOW_DOMAIN . '_follow_links'][$key]['name']);
+      $row[] = drupal_render($form[FOLLOW_DOMAIN . '_follow_links'][$key]['url']);
       if (user_access('change follow link titles')) {
-        $row[] = drupal_render($form['follow_links'][$key]['title']);
+        $row[] = drupal_render($form[FOLLOW_DOMAIN . '_follow_links'][$key]['title']);
       }
 
       // Now, render the weight row.
-      $form['follow_links'][$key]['weight']['#attributes']['class'][] = 'follow-links-weight';
-      $row[] = drupal_render($form['follow_links'][$key]['weight']);
+      $form[FOLLOW_DOMAIN . '_follow_links'][$key]['weight']['#attributes']['class'] = 'follow-links-weight';
+      $row[] = drupal_render($form[FOLLOW_DOMAIN . '_follow_links'][$key]['weight']);
 
       // Add the new row to our collection of rows, and give it the 'draggable' class.
       $rows[] = array(
@@ -586,7 +587,7 @@ function theme_follow_links_form($variables) {
       );
     }
     else {
-      $disabled_rows[] = array(drupal_render($form['follow_links'][$key]['name']), drupal_render($form['follow_links'][$key]['url']));
+      $disabled_rows[] = array(drupal_render($form[FOLLOW_DOMAIN . '_follow_links'][$key]['name']), drupal_render($form[FOLLOW_DOMAIN . '_follow_links'][$key]['url']));
     }
   }
 
